lib/matplotlib/backend_bases.py

```diff
@@ -1546,8 +1546,10 @@ class Done(Exception):
     def _draw(renderer): raise Done(renderer)
 
     with cbook._setattr_cm(figure, draw=_draw):
+        print_method = getattr(
+            figure.canvas._get_output_canvas(fmt), f"print_{fmt}")
         try:
-            print_method(io.BytesIO())
```

Comment:
`dpi` was many other things than `figure.dpi` in my quick skim of the code below, so is it really OK to use `figure.dpi` here?  I'll block because I think this should be answered, but anyone should clear if this is really not a problem.  I haven't looked carefully yet.

Reply:
If _get_renderer is called from print_figure(), then figure.dpi has already been set to match the requested savefig dpi (`cbook._setattr_cm(self.figure, dpi=dpi)`), so setting it a second time is a noop.  If called from tight_layout.get_renderer, OTOH, we do need to pass in the figure dpi -- otherwise it's savefig.dpi which gets used, which is not correct when we want to tight-layout something not for the purpose of saving it.